src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -243,16 +248,41 @@ uct_cuda_ipc_mkey_pack(uct_md_h md, uct_mem_h tl_memh, void *address,
 {
     uct_cuda_ipc_rkey_t *packed = mkey_buffer;
     uct_cuda_ipc_memh_t *memh   = tl_memh;
-    uct_cuda_ipc_lkey_t *key;
+    uct_cuda_ipc_lkey_t *key, *tmp;
     ucs_status_t status;
+    unsigned long long buffer_id;
 
-    ucs_list_for_each(key, &memh->list, link) {
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuPointerGetAttribute(&buffer_id,
+                                      CU_POINTER_ATTRIBUTE_BUFFER_ID,
+                                      (CUdeviceptr)address));
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    ucs_list_for_each_safe(key, tmp, &memh->list, link) {
         if (((uintptr_t)address >= key->d_bptr) &&
             ((uintptr_t)address < (key->d_bptr + key->b_len))) {
-            goto found;
+            ucs_trace("found range (%p ... %p) in local cache", address,
+                      address + length);
+            if (buffer_id == key->ph.buffer_id) {
```

Comment:
can remove this logic since rcache evicts memh if memory is released (using memory hooks invalidate logic)